src/main/scala/org/bitcoins/core/util/Base58.scala

```diff
@@ -55,6 +60,67 @@ trait Base58 extends BitcoinSLogger {
     }
   }
 
+  def encode(hex : String) : String = {
+    val bytes = BitcoinSUtil.decodeHex(hex)
+    encode(bytes)
+  }
+
+  def encode(byte : Byte) : String = encode(Seq(byte))
+
+  /**
+    * Encodes a Base58 address from a hash
+    * @param hash The result of Sha256(RipeMD-160(public key))
+    * @param addressType string. Either "pubkey" or "script"
+    * @param isTestNet boolean
+    * @return
+    */
+  def encodePubKeyHashToBase58Address(hash: Sha256Hash160Digest,
+                                  addressType : String,
+                                  isTestNet : Boolean) : String = {
+    val versionByte : Byte = {
+      require(addressType != "pubkey" || addressType != "script",
+        throw new IllegalArgumentException("Address must be of type 'pubkey' or 'script'."))
+      if (!isTestNet) {
+        if (addressType == "pubkey") MainNet.p2pkhNetworkByte
+        else MainNet.p2shNetworkByte
+      }
+      else if (isTestNet) {
+        if (addressType == "pubkey") TestNet3.p2pkhNetworkByte
+        else TestNet3.p2shNetworkByte
+      }
+      else throw new IllegalArgumentException("Something broke -- Check your parameters. There should be a " +
```

Comment:
This case can never happen, since if(!isTestNet), and else if(isTestNet) are inverses, so one will always evaluate to true. This is dead code